### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by PetroczyP**
This pull request adds major new features and improvements to the Mergington High School extracurricular activities management system. The backend API is enhanced with the ability to unregister students from activities, and the frontend now allows for participant management with a user-friendly interface. Comprehensive automated tests have also been introduced to ensure reliability. The codebase benefits from improved documentation and maintainability.

**Backend API improvements:**

* Added a new `DELETE /activities/{activity_name}/unregister` endpoint to allow students to unregister from activities, with appropriate error handling and documentation. (`src/app.py` [src/app.pyR56-R201](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00R56-R201))
* Improved and expanded docstrings for all API endpoints, clarifying arguments, return values, and error cases. (`src/app.py` [[1]](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00L1-R38) [[2]](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00R56-R201)
* Expanded the in-memory `activities` database with several new example activities for demonstration and testing purposes. (`src/app.py` [src/app.pyR56-R201](diffhunk://#diff-04791d82dd15fdd480f084d7ef65a10789fa5012cb7935f76080763444d48a00R56-R201))

**Frontend enhancements:**

* Updated the UI to display current participants for each activity, including a styled list and a "Delete" button for each participant. (`src/static/app.js` [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R23-R42) `src/static/styles.css` [[2]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R77-R131)
* Implemented frontend logic to handle participant removal via the new unregister API, including confirmation dialogs and success/error messaging. (`src/static/app.js` [src/static/app.jsR99-R144](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R99-R144))

**Testing and quality:**

* Added a comprehensive test suite using pytest to cover all API endpoints, including signup, unregistration, and edge cases, ensuring robust behavior. (`tests/test_app.py` [tests/test_app.pyR1-R201](diffhunk://#diff-a67cb1853203a6f1956991a9d9881d231c4d43557f5baffd45cc672a87e41cc6R1-R201))
* Added a module docstring to the test package for clarity. (`tests/__init__.py` [tests/__init__.pyR1](diffhunk://#diff-001e61d97d9dee27d0c5aa1f23ba6c1972cd8a2e8320bac839656b0ab935b84dR1))

### CODE REVIEW

File: src/static/app.js [hunk 1]
```diff
@@ -20,11 +20,26 @@ document.addEventListener("DOMContentLoaded", () => {
...
+                </li>
+              `).join('')}
+             </ul>`
+          : `<p class="no-participants">No participants yet. Be the first to sign up!</p>`;
+
         activityCard.innerHTML = `
           <h4>${name}</h4>
           <p>${details.description}</p>
           <p><strong>Schedule:</strong> ${details.schedule}</p>
           <p><strong>Availability:</strong> ${spotsLeft} spots left</p>
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply src/static/app.js:38]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/PetroczyP/skills-getting-started-with-github-copilot/pull/2#discussion_r2635847280)

File: src/app.py [hunk 2]
```diff
@@ -38,30 +53,149 @@
...
+            detail="Student already signed up for this activity"
+        )
+
+    # Add student to the activity's participant list
     activity["participants"].append(email)
     return {"message": f"Signed up {email} for {activity_name}"}
+
+
+@app.delete("/activities/{activity_name}/unregister")
+def unregister_from_activity(activity_name: str, email: str) -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply src/app.py:173]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/PetroczyP/skills-getting-started-with-github-copilot/pull/2#discussion_r2635847291)

File: tests/test_app.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        )
+        assert response2.status_code == 200
+        
+        # Verify both students are registered
+        activities_response = client.get("/activities")
+        activities_data = activities_response.json()
+        participants = activities_data["Programming Class"]["participants"]
+        assert "student1@mergington.edu" in participants
+        assert "student2@mergington.edu" in participants
+
```
[the remark this answers is not in the document]
Comment (PetroczyP) [reply tests/test_app.py:147]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/PetroczyP/skills-getting-started-with-github-copilot/pull/2#discussion_r2635847312)

**Comment by PetroczyP** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/PetroczyP/skills-getting-started-with-github-copilot/pull/2#pullrequestreview-3599508414)
